lib/slave/src/ESMStates.cc

```diff
@@ -0,0 +1,238 @@
+        {
+            // If in INIT and asked for INIT, clear the error
+            if (control.get_requested_state() == INIT)
+            {
+                return Context::build(INIT);
+            }
+
+            // If didn't asked asked for INIT stay where it is and don't do anything
+            return Context::build(currentStatus.al_status, currentStatus.al_status_code);
+        }
+
+        // Unknown state request
+        auto requestedState = control.get_requested_state();
+        if (currentStatus.get_state() != State::OPERATIONAL and requestedState != State::BOOT
+            and requestedState != State::INIT and requestedState != State::PRE_OP and requestedState != State::SAFE_OP
+            and requestedState != State::OPERATIONAL)
+        {
+            return Context::build(id_, StatusCode::UNKNOWN_REQUESTED_STATE);
+        }
+
+        return routine_internal(currentStatus, control);
+    }
+
+    Context Init::routine_internal(Context, ALControl control)
+    {
+        if (control.get_requested_state() == State::PRE_OP)
+        {
+            if (not mbx_)
+            {
+                return Context::build(State::PRE_OP);
+            }
+            if (mbx_->configureSm() == hresult::OK)
+            {
+                if (mbx_->is_sm_config_ok())
+                {
+                    return Context::build(State::PRE_OP);
+                }
+                else
+                {
+                    return Context::build(State::INIT, StatusCode::INVALID_MAILBOX_CONFIGURATION_PREOP);
+                }
+            }
+        }
+
+        // Invalid state request
+        if ((control.get_requested_state() == State::SAFE_OP) or (control.get_requested_state() == State::OPERATIONAL))
+        {
+            return Context::build(State::INIT, StatusCode::INVALID_REQUESTED_STATE_CHANGE);
+        }
+
+        // BOOTSTRAP not supported yet. If implemented, need to check the SII to know if enabled.
+        if (control.get_requested_state() == State::BOOT)
+        {
+            return Context::build(id_, StatusCode::BOOTSTRAP_NOT_SUPPORTED);
+        }
+
+        return Context::build(State::INIT);
+    }
+
+    PreOP::PreOP(AbstractESC& esc, PDO& pdo)
+        : AbstractState(State::PRE_OP, esc, pdo)
+    {
+    }
+
+    void PreOP::on_entry(Context, Context)
+    {
+        if (mbx_)
+        {
+            mbx_->set_sm_activate(true);
+        }
+        pdo_.set_sm_output_activated(false);
+        pdo_.set_sm_input_activated(false);
+    }
+
+    Context PreOP::routine_internal(Context, ALControl control)
+    {
+        if (mbx_ and not mbx_->is_sm_config_ok())
+        {
+            return Context::build(State::INIT, StatusCode::INVALID_MAILBOX_CONFIGURATION_PREOP);
+        }
+
+        if (control.get_requested_state() == State::SAFE_OP)
+        {
+            if (pdo_.configure_pdo_sm() != hresult::OK)
+            {
+                return Context::build(id_);
```

Comment:
No Status code here ? asked for SafeOP and pdoSM not Ok ?

Reply:
no, no error, we just wait for the master to configure the sync managers